### PR TITLE
add hoc2015 blockly levels to edit-code script

### PR DESCRIPTION
**Pull request by cpirich**
- Added 15 blockly hoc2015 levels as a stage on the existing edit-code script (for testing only)
- Actually created the first 6 blockly hoc2015 levels in levels.js - setting startBlocks, toolbox, and requiredBlocks for each

### CODE REVIEW

File: apps/src/studio/levels.js [hunk 1]
```diff
@@ -54,6 +54,64 @@ function saySpriteRequiredBlock(options) {
 }
 
 /**
+ * Constructs a required block definition to match "move [sprite] [dir]" blocks
+ * @param options (all optional):
```
Comment (breville) [inline apps/src/studio/levels.js:58]: Is this the standard documentation format?  I've used something like this that I found elsewhere in our code..
- @returns {Object} outcome
- @returns {boolean} outcome.exists Whether the level even has any of these criteria.
- @returns {boolean} outcome.achieved false if any of the criteria was required but not met.
- @returns {string} outcome.message A custom message for the first of the failing criteria.

File: apps/src/studio/levels.js [hunk 2]
```diff
@@ -2528,3 +2525,65 @@ levels.js_hoc2015_event_free = {
     }
   ],
 };
+
+levels.hoc2015_blockly_1 = utils.extend(levels.js_hoc2015_move_right,  {
+  editCode: false,
+  startBlocks: whenRunMoveEast,
+  toolbox: tb(hocMoveNSEW),
+  requiredBlocks: [
+    moveRequiredBlock({ dir: '2' }), // East
```
Comment (breville) [inline apps/src/studio/levels.js:2534]: would it be easy to use readable constants for '2' etc.?

File: apps/src/studio/levels.js [hunk 3]
```diff
@@ -54,6 +54,64 @@ function saySpriteRequiredBlock(options) {
 }
 
 /**
+ * Constructs a required block definition to match "move [sprite] [dir]" blocks
+ * @param options (all optional):
+ *          sprite (string): zero-indexed string ID of sprite, e.g., "1"
+ *          dir (string): string of Direction constant. We show
+ *            the direction in feedback blocks
+ * @returns test definition suitable for feedback.js::getMissingRequiredBlocks
+ *          required block processing
```
Comment (islemaster) [inline apps/src/studio/levels.js:63]: Nonblocking: Could JSDoc it up even more!

``` javascript
/**
 * Constructs a required block definition to match "move [sprite] [dir]" blocks
 * @param {string} [options.sprite] zero-indexed string ID of sprite, e.g., "1"
 * @param {string} [options.dir] string of Direction constant. We show
 *        the direction in feedback blocks
 * @returns {Array} test definition suitable for getMissingRequiredBlocks
 *          required block processing
 * @see FeedbackUtils#getMissingRequiredBlocks
 */
```

File: apps/src/studio/levels.js [hunk 4]
```diff
@@ -54,6 +54,64 @@ function saySpriteRequiredBlock(options) {
...
+  return [
+    {
+      test: function (block) {
+        if (block.type !== 'studio_move') {
+          return false;
+        }
+        if (options.sprite && block.getTitleValue("SPRITE") !== options.sprite) {
+          return false;
+        }
+        if (options.dir && block.getTitleValue("DIR") !== options.dir) {
```
Comment (islemaster) [inline apps/src/studio/levels.js:83]: Do "SPRITE" and "DIR" exist as constants anywhere?

File: apps/src/studio/levels.js [hunk 5]
```diff
@@ -54,6 +54,64 @@ function saySpriteRequiredBlock(options) {
...
+  <block type="studio_move"><title name="DIR">2</title></block> \
+  <block type="studio_move"><title name="DIR">8</title></block>';
+
+var whenRunMoveEast = '<block type="when_run"><next> \
+  <block type="studio_move"><title name="DIR">2</title></block></next> \
+  </block>';
+
+var whenRunMoveSouth = '<block type="when_run"><next> \
+  <block type="studio_move"><title name="DIR">4</title></block></next> \
+  </block>';
```
Comment (islemaster) [inline apps/src/studio/levels.js:112]: :+1: Reusable stuff like this is great!

File: apps/src/studio/levels.js (hunk 2 above)
Comment (islemaster) [inline apps/src/studio/levels.js:2534]: Why aren't we loading and using the `Direction` constants here?

**Comment by islemaster** [comment]
LGTM.  Sorry Chris for making all the same comments as Brendan!

File: apps/src/studio/levels.js (hunk 4 above)
Comment (cpirich) [inline apps/src/studio/levels.js:83]: Not yet, we may want to do that going forward. Maybe a blockConstants.js file?

File: apps/src/studio/levels.js (hunk 2 above)
Comment (cpirich) [inline apps/src/studio/levels.js:2534]: Because they are string. Will work on improving this further once we settle on NSEW or UDLR

**Pull request merged**